jung-algorithms/src/main/java/edu/uci/ics/jung/layout/algorithms/BalloonLayoutAlgorithm.java

```diff
@@ -87,7 +85,11 @@ protected void setRootPolar(LayoutModel<N> layoutModel, N root) {
   }
 
   protected void setPolars(
-      LayoutModel<N> layoutModel, Set<N> kids, Point parentLocation, double parentRadius) {
+      LayoutModel<N> layoutModel,
+      Set<N> kids,
+      Point parentLocation,
+      Point grandparentLocation,
+      double parentRadius) {
 
```

Comment:
This 'parent' and 'grandparent' stuff is for computing and tracking the angle so I can use it to compute the value to bisect the angle between child nodes. This is a (working) prototype, subject to further refactoring. FWIW, The layout is prettier now.